src/NumberInput/NumberInput.svelte

```diff
@@ -135,7 +135,7 @@
   $: dispatch("change", value);
   $: incrementLabel = translateWithId("increment");
   $: decrementLabel = translateWithId("decrement");
-  $: value = inputValue != null ? Number(inputValue) : null;
+  $: value = inputValue != "" ? Number(inputValue) : null;
```

Comment:
A couple of things:

1. A strict inequality check is needed here.

Otherwise, a value of `0` will be converted to `null`.

```
0 != ""; // false
0 !== ""; // true
```

2. We still need a null check to account for the following cases

This should render an empty value with an error outline:

```svelte
<NumberInput value={null} />
```

This should render an empty value without an error:

```svelte
<NumberInput allowEmpty value={null} />
```

```suggestion
  $: value = inputValue !== "" && inputValue != null ? Number(inputValue) : null;
```

Reply:
Ah, yes, because there is the initial assignment `inputValue = value`, so the variable could contain `string`, `number` and `null`. #1053 no longer has this issue.